Src/FluentAssertions/Primitives/StringAssertions.cs

```diff
@@ -821,19 +1101,19 @@ public AndConstraint<TAssertions> NotEndWith(string unexpected, string because =
     {
         Guard.ThrowIfArgumentIsNull(unexpected, nameof(unexpected), "Cannot compare end of string with <null>.");
 
-        bool success = Execute.Assertion
-            .ForCondition(Subject is not null)
-            .BecauseOf(because, becauseArgs)
-            .FailWith("Expected {context:string} that does not end with {1}{reason}, but found {0}.", Subject, unexpected);
+        bool notEquivalent;
 
-        if (success)
+        using (var scope = new AssertionScope())
         {
-            Execute.Assertion
-                .ForCondition(!Subject.EndsWith(unexpected, StringComparison.Ordinal))
-                .BecauseOf(because, becauseArgs)
-                .FailWith("Expected {context:string} {0} not to end with {1}{reason}.", Subject, unexpected);
+            Subject.Should().EndWith(unexpected);
+            notEquivalent = scope.Discard().Length > 0;
         }
 
+        Execute.Assertion
+            .ForCondition(Subject != null && notEquivalent)
+            .BecauseOf(because, becauseArgs)
+            .FailWith("Expected {context:string} not to end with {0}{reason}, but found {1}.", unexpected, Subject);
```

Comment:
I agree that this phrasing is better than the "that does not" 👍 
Will you also adjust the four remaining occurrences of "that does"?

Reply:
You are right, that would be more consistent. I replaced the five remaining occurences of "that does not" with "not to".